custom/cpurand/src/lib.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2018 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+#![no_std]
+
+#[macro_use]
+extern crate cfg_if;
+
+use getrandom::{register_custom_getrandom, Error};
+
+cfg_if! {
+    if #[cfg(target_arch = "x86_64")] {
+        #[path = "../../../src/util.rs"] mod util;
+        #[path = "../../../src/rdrand.rs"] mod imp;
```

Comment:
Do we want to use out-of-crate modules? We do have a reasonable option to resolve this in the future (copy `util`, move `rdrand`), so I guess it's acceptable, though probably any use of `#[path]` on modules is bad style.

Alternatively for `util` we could export `LazyBool` under some feature flag.

Reply:
I mean we do use `#[path]` a ton in the main `src/lib.rs`, but I get your point. 

I wasn't able to figure out how to structure the code to avoid the use of out-of-crate modules completely. Feature flags could solve this, and I even tried this initially, but it made the code much messier and harder to understand.

Do you have a way you would like this to be organized? The existing layout isn't _great_, but I couldn't figure out a better way.

Reply:
I guess it's not really important.

The plan is to remove `rdrand` from `getrandom` core, right? Or were you planning on keeping it for SGX support within `getrandom`? In the latter case, there seems to be less reason to make `getrandom-cpu` a separate crate instead of merely a feature flag.

The proposal in #84 appears to be somewhat different: using the `getrandom-cpu` crate directly, but renamed `getrandom`. This of course has the limitation that it only applies to the current user of `getrandom` (i.e. won't fix crates importing `rand` like usual). Perhaps best to take this back to #84...

Reply:
This is fixed. Now, the only use of out-of-crate modules are the custom RNG integration tests.